pypdf/_page.py

```diff
@@ -1519,6 +1519,23 @@ def compressContentStreams(self) -> None:  # deprecated
         )
         self.compress_content_streams()
 
+    @property
+    def page_number(self) -> int:
+        """
+        Read-only property which return the page number with the pdf file.
+
+        Returns:
+            int : page number ; -1 if the page is not attached to a pdf
+        """
+        if self.indirect_reference is None:
+            return -1
+        else:
+            try:
+                lst = self.indirect_reference.pdf.pages
+                return lst.index(self)
+            except ValueError:
+                return -1
```

Comment:
```suggestion
    def page_number(self) -> Optional[int]:
        """
        Read-only property which return the page number with the pdf file.

        Returns:
            page number ; if the page is not attached to a PDF, None is returned
        """
        if self.indirect_reference is None:
            return None
        else:
            try:
                lst = self.indirect_reference.pdf.pages
                return lst.index(self)
            except ValueError:
                return None
```

Reply:
@pubpub-zz Would it be OK for you to do it like this?

I want to avoid having a special integer value for "not attached to a PDF", because this typically leads easily to silent errors for users. Returning `None` makes sure that mypy will ensure the users is taking care of that special case.